packages/components/psammead-social-embed/src/Canonical/useScript.js

```diff
@@ -0,0 +1,25 @@
+import { useEffect } from 'react';
+
+/**
+ * useScript is a custom hook that appends a non-blocking script element
+ * to the head if it doesn't already exist and removes it in clean-up.
+ * @param {string} src The URL of the script to be appended.
+ */
+const useScript = src =>
+  // eslint-disable-next-line consistent-return
+  useEffect(() => {
+    const hasScript = !!document.querySelector(`head script[src="${src}"]`);
+
+    if (src && !hasScript) {
+      const script = document.createElement('script');
+      script.src = src;
+      script.async = true;
+      document.head.appendChild(script);
+
+      return () => {
```

Comment:
Just for my understanding, this cleanup function is only returned when `src` exists and the script doesn't exist?

Reply:
The hook will execute when, as you say, "src exists and the script doesn't exist". When it executes, it also returns that function which React calls when the component unmounts. The condition doesn't prevent this from running.

Reply:
Ahh okay cool. Would it affect anything to have this outside of the conditional to not cause any confusion?

Reply:
If we were to remove the clean-up function outside of the conditional, it wouldn't have anything to clean up, because the script wouldn't have been added.